wyvern/components/models/model_component.py

```diff
@@ -112,6 +112,7 @@ async def execute(self, input: INPUT_TYPE, **kwargs) -> MODEL_OUTPUT:
         def events_generator() -> List[ModelEvent]:
             timestamp = datetime.utcnow()
             all_events: List[ModelEvent] = []
+            model_name = model_output.model_name or self.name or self.__class__.__name__
```

Comment:
But: it feels error prone to keep adding cases here. Should we default self.name instead?

Reply:
yeah, i can deprecate model_name in model_output.

Reply:
yeah, i don't like the design of `model_name under model_output`

Reply:
@suchintan updated.